File: 8kyu/(8 kyu) Remove First and Last Character/(8 kyu) Remove First and Last Character.cs
using System;

// 1 Simple solution
public class Kata
{
    public static string Remove_char(string s)
    {
        string result = "";

        for (int i = 1; i < s.Length - 1; i++)
        {
            result += s[i];
        }

        return result;
    }
}

// 2 Optimized solution
public class Kata
{
    public static string Remove_char(string s)
    {
        return s.Substring(1, s.Length - 2);
    }
}

// 3 Coding golf
public class Kata
{
    public static string Remove_char(string s) => s[1..^1];
}

File: 8kyu/(8 kyu) Remove First and Last Character/(8 kyu) Remove First and Last Character.py
# 1 Simple solution
def remove_char(s):
    result = ""
    for i in range(1, len(s) - 1):
        result += s[i]
    return result

# 2 Optimized solution
def remove_char(s):
    if len(s) <= 2:
        return ""
    return s[1:-1]

# 3 Clever solution
def remove_char(s):
    return s[1:-1] if len(s) > 2 else ""

# 4 Coding golf
remove_char=lambda s:s[1:-1]